### ISSUE TITLE
[PHP 8] Warning on EncodesUrl

### ISSUE THREAD

**Issue by mattkingshott** [thephpleague/oauth1-client#127]
The following PHP warning is generated under PHP 8:

```php
League/OAuth1/Client/Signature/HmacSha1Signature::League/OAuth1/Client/Signature/{closure}(): 
Argument #2 ($value) must be passed by reference, value given in 
/vendor/league/oauth1-client/src/Signature/EncodesUrl.php on line 54
```

Seems to be caused by the second parameter being passed as a reference. Removing the '&' symbol from value seems to work for me, but I don't know if there are greater implications outside of my particular use case.

**Comment by mattkingshott** [comment]
Submitted PR #128 to fix this.

**Comment by bencorlett** [comment]
Thanks Matt! I’ll take a look 👍

**Comment by smares** [comment]
Submitted https://github.com/thephpleague/oauth1-client/pull/129 to not only get rid of the exception, but actually make the code do what it was supposed to do - that is, normalize keys and values. First time doing a pull request, so I hope I didn't do any mistakes. 😅

**Comment by mattkingshott** [comment]
Great. Thanks!

**Comment by mattkingshott** [comment]
@bencorlett would you mind taking a look? Thanks!

**Comment by smares** [comment]
Any idea when a new release is going to be issued? Laravel Socialite is using this library and currently the Twitter Login is broken on PHP 8 because of this bug. :(

**Comment by dimitri-koenig** [comment]
@bencorlett: Could you please create a new version since this bug is in the current released version 1.8.2 which is not usable due to this bug. That would be great.

**Comment by bencorlett** [comment]
I anticipate that I’ll have 1.9.0 out within a day or two. Thank you for your patience.

**Comment by dimitri-koenig** [comment]
Sounds great, thx a lot

**Comment by mattkingshott** [comment]
@bencorlett any update on getting 1.9.0 out the door? Sorry to be a nuisance. Thanks!

**Comment by bencorlett** [comment]
I am away for the Christmas holidays. Will be early in the new year. Thanks

**Comment by mattkingshott** [comment]
@bencorlett really sorry to be a nuisance buddy, but is there any chance of 1.9.0 getting tagged this week as I'm hoping to push a major new app version next week that uses this library with PHP 8?

**Comment by smares** [comment]
Because of issues like this in the past, we ended up creating a GitHub organization account where we fork all repositories we use in our projects. You can then decide whether to always use the own fork or only use the fork when a fix is taking too long in upstream.

**Comment by bencorlett** [comment]
@mattkingshott thanks for your interest, and patience in the project mate :) I haven't been committing too much time to OSS over the past couple of weeks but rather spending quality time away from the computer with my family.

There's a few moving parts to getting the next release out. Firstly, we need to get [this PR](https://github.com/thephpleague/oauth1-client/pull/132) to build so that we can merge it in. There's a few prerequisites for that, namely working around [php-cs-fixer's lack of PHP 8 support](https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4702) and also hacking past PHPStan's errors regarding OpenSSL changes in PHP 8, such as [this](https://github.com/thephpleague/oauth1-client/pull/132/commits/1632e3fa569e146e6f46f284b4483429657e48be) and [this](https://github.com/thephpleague/oauth1-client/pull/132/commits/b2850cd03cc5e6e8285ae91fafe838f78bb58983).

Given the large user base of this library, both directly and through Laravel packages, it's imperative that the release goes smoothly. The additional checks and balances performed by analysis tooling are there to pick up potential issues, so any workaround for their support for PHP 8 must be carefully considered.

I'm currently camping with limited internet access however I'm prioritising getting [this PR](https://github.com/thephpleague/oauth1-client/pull/132) to build so I can merge and release it.

@smares' suggestion is something I regularly do as well - you can read more on that [here](https://getcomposer.org/doc/05-repositories.md#loading-a-package-from-a-vcs-repository) :) 

Thanks!

**Comment by mattkingshott** [comment]
@bencorlett no worries mate. I've made a fork for now. 

Enjoy time off with the family :)

**Comment by keradus** [comment]
> namely working around php-cs-fixer's lack of PHP 8 support

you can use PHP CS Fixer 2.18+ for PHP 8

**Comment by bencorlett** [comment]
> > namely working around php-cs-fixer's lack of PHP 8 support
> 
> you can use PHP CS Fixer 2.18+ for PHP 8

Wonderful! I saw they released this a couple of days back. I've got a workaround in place and I have published v1.9.0, but I'll look to upgrade this at some point.
